rcgen/examples/rsa-irc-openssl.rs

```diff
@@ -1,5 +1,3 @@
-#![allow(clippy::complexity, clippy::style, clippy::pedantic)]
```

Comment:
Not sure this tracks with your commit message? If new lints appear in any of these categories we're still going to warn about these, which (according to this `#![allow]`) is unwanted?

Reply:
I thought being warned for new lints that apply would be the desired behaviour.

Reply:
I reworked the commit message.

Reply:
I read this `#![allow]` as wanting to not see warnings for new lints from these particular categories.

Reply:
I guess I'm not sure why you wouldn't want to see those warnings. Fixing them doesn't seem like a tremendous burden and will keep the example code more idiomatic.

If it's a point of contention I can drop this change but it seems like an unnecessary concern.

Reply:
(I'm not saying having this would be my choice... just guessing at the original intention.)

Reply:
Let's wait and see what est31 thinks. I'll restore the `allow`'s if requested, but if that's the direction we go it should be done consistently. `sign-leaf-with-ca.rs` is missing the `allow` and should be fixed.